lib/server.js

```
@@ -1,32 +1,54 @@
+/**
+ * Dependencies
+ */
 var socket = require('./sockets/rep');
 var log = require('log');
+var utils = require('./utils');
 
-function MicroServiceServer(name, port){
+function MicroServiceServer(name, addr){
   var self = this;
   this.name = name;
+  addr = utils.parseSocketAddress(addr);
   this.logger = log(name + ':server');
-  this.logger.info('binding to port ' + port);
+  this.logger.info('binding to addr ' + addr);
 
-  this.sock = new socket().bind(port);
+  this.sock = new socket().bind(addr);
 
   this.sock.on('bind',function(){
-    self.logger.info('bound to port ' + port);
+    self.logger.info('bound to addr ' + addr);
   });
 
   this.sock.on('connect',function(){
-    self.logger.info('connected :' + port);
+    self.logger.info('connected :' + addr);
     self.connect();
   });
 
   this.sock.on('disconnect',function(){
-    self.logger.info('connection closed :' + port);
+    self.logger.info('connection closed :' + addr);
     self.disconnect();
   });
 
   this.sock.on('message', function (serviceJob ,cb) {
     if(serviceJob.server === self.name) {
-      if (serviceJob.reply && typeof cb === 'function')  serviceJob.arguments.push(cb);
-      if (serviceJob.method && typeof self[serviceJob.method] === 'function') {
+      var serverMethod = serviceJob.method && typeof self[serviceJob.method] === 'function' ? self[serviceJob.method] : null;
+      //check if clientCall was requested with a callback
+      if (serviceJob.reply && typeof cb === 'function'){
+        //if serverMethod set up as req/rep then add to end of arguments
+        if(serverMethod._callbacks) serviceJob.arguments.push(cb);
+        else{
+          //if set up as pushPull reply with no data
+          log.warning('Client provided a callback, called a method that does not expect one.')
+          cb();
+        }
+      }
+      else{
+        if(serverMethod._callbacks){
+          log.warning('Client did not provide a callback, called a method that expected one.')
```

The issue with failing silently is that a function is cached on the client side waiting for a response